src/Integration.Vsix/Analysis/UserSettingsProvider.cs

```diff
@@ -18,112 +18,137 @@
  * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
  */
 
-using System;
 using System.ComponentModel.Composition;
-using System.Diagnostics;
 using System.IO;
 using System.IO.Abstractions;
 using SonarLint.VisualStudio.Core;
+using SonarLint.VisualStudio.SLCore;
+using SonarLint.VisualStudio.SLCore.Core;
+using SonarLint.VisualStudio.SLCore.Service.Rules;
 
-namespace SonarLint.VisualStudio.Integration.Vsix.Analysis
+namespace SonarLint.VisualStudio.Integration.Vsix.Analysis;
+
+[Export(typeof(IUserSettingsProvider))]
+internal sealed class UserSettingsProvider : IUserSettingsProvider, IDisposable
 {
```

Comment:
Git diff is so broken because of the namespace change...

Reply:
Sorry for that! I wanted to apply CaYC and use file scoped namespace for the classes that I touch. Maybe the commit diff is more helpful?